### PR TITLE
support for accessing artifacts and their metadata

### PR DESCRIPTION
**Pull request by volkc-basf**
With this PR I'm adding support for the _FileInfo_ endpoint. Furthermore there's now a _FileContents_ function that can be used to download files stored in the artifactory.

Reason is, that we'd like to extend the artifactory terraform provider.
https://github.com/atlassian/terraform-provider-artifactory/issues/63

Please let me know if you do have further questions.

Best,
Christian

### CODE REVIEW

File: artifactory/v1/artifacts.go [hunk 1]
```diff
@@ -183,3 +184,63 @@ func (s *ArtifactService) DeleteRepositoryReplicationConfig(ctx context.Context,
...
+	fileInfo := new(FileInfo)
+	resp, err := s.client.Do(ctx, req, fileInfo)
+	return fileInfo, resp, err
+}
+
+// Description: Copies the specified file to the given target. Supported by local, local-cached and virtual repositories.
+// Security: Requires a privileged user (can be anonymous)
+func (s *ArtifactService) FileContents(ctx context.Context, repoKey string, filePath string, target interface{}) (*FileInfo, *http.Response, error) {
+	if target == nil {
+		return nil, nil, fmt.Errorf("Target is not allowed to be nil")
```
Comment (jamestoyer) [inline artifactory/v1/artifacts.go]: Could you lowercase `Target` to follow the Golang conventions

File: artifactory/v1/artifacts.go [hunk 2]
```diff
@@ -183,3 +184,63 @@ func (s *ArtifactService) DeleteRepositoryReplicationConfig(ctx context.Context,
...
+// Description: Returns the metadata of the given file. Supported by local, local-cached and virtual repositories.
+// Security: Requires a privileged user (can be anonymous)
+func (s *ArtifactService) FileInfo(ctx context.Context, repoKey string, filePath string) (*FileInfo, *http.Response, error) {
+	path := fmt.Sprintf("/api/storage/%s/%s", repoKey, filePath)
+	req, err := s.client.NewRequest("GET", path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req.Header.Set("Accept", client.MediaTypeJson)
```
Comment (jamestoyer) [inline artifactory/v1/artifacts.go]: I don't believe this is the correct type. You will want to create a constant in `types.go` and use `application/vnd.org.jfrog.artifactory.storage.FileInfo+json` for the value as per the [docs](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-FileInfo). This is the convention we've used elsewhere. This also allows you to get rid of the import for `"github.com/atlassian/go-artifactory/v2/artifactory/client"`

**Review by jamestoyer — CHANGES_REQUESTED** [review]

File: artifactory/v1/artifacts.go (hunk 2 above)
Comment (volkc-basf) [reply artifactory/v1/artifacts.go]: Fixed with bb293b1

File: artifactory/v1/artifacts.go (hunk 1 above)
Comment (volkc-basf) [reply artifactory/v1/artifacts.go]: Fixed with 058d926

**Review by jamestoyer — APPROVED** [review]

**Pull request merged**